### ISSUE TITLE
Linter emits "Failed to download module" errors when module inspection is disabled

### ISSUE THREAD

**Issue by arcjor** [terraform-linters/tflint#1675]
### Summary

When module inspection is disabled there is no value in downloading modules.

If private modules are referenced in source but the CI system is not instrumented to download them then the log will be flooded with errors. Since there is no intent to download these modules the linter should handle these failures to download silently, as they are not erroneous.

```
╷
│ Error: Failed to download module
│ 
│ Could not download module "tag_generator" (main.tf:9) source code from
│ "git::https://github.com/myorg/terraform-azure-modules.git?ref=myref":
│ error downloading
│ 'https://github.com/myorg/terraform-azure-modules.git?ref=myref':
│ /usr/bin/git exited with 128: Cloning into
│ '.terraform/modules/tag_generator'...
│ fatal: could not read Username for 'https://github.com/': No such device or
│ address
│ 
╵

╷
│ Error: Failed to download module
│
│ Could not download module "default_tagging_policy_set" (infra/main.tf:90)
│ source code from
│ "git::ssh://git@github.com/myorg/terraform-azure-modules.git?ref=myref":
│ error downloading
│ 'ssh://git@github.com/myorg/terraform-azure-modules.git?ref=myref':
│ /usr/bin/git exited with 128: Cloning into
│ '.terraform/modules/infra.default_tagging_policy_set'...
│ Host key verification failed.
│ fatal: Could not read from remote repository.
│
│ Please make sure you have the correct access rights
│ and the repository exists.
│
╵
```

### Command

tflint

### Terraform Configuration

```terraform
module "linux_virtual_machine" {
  source = "git@github.com:myorg/terraform-azure-modules.git//modules/Azure/AzureRM-Compute-virtualMachines?ref=myref"

  naming_prefix       = module.naming.vm_prefix
  location            = module.rg_and_tags.location
  resource_group_name = module.rg_and_tags.name
  os                  = "Linux"
  key_vault_id        = module.key_vault.id
  subnet_id           = module.networking.subnet_ids["VMSubnet"]
  vm_size             = "Standard_B2ms"
  identity_ids        = [module.deploy_managed_identity.id]
  tags                = module.rg_and_tags.tags
}
```

### TFLint Configuration

```HCL
config {
  module = false
  force = false
}
```

### Output

```Shell
╷
│ Error: Failed to download module
│ 
│ Could not download module "tag_generator" (main.tf:9) source code from
│ "git::https://github.com/myorg/terraform-azure-modules.git?ref=myref":
│ error downloading
│ 'https://github.com/myorg/terraform-azure-modules.git?ref=myref':
│ /usr/bin/git exited with 128: Cloning into
│ '.terraform/modules/tag_generator'...
│ fatal: could not read Username for 'https://github.com/': No such device or
│ address
│ 
╵

╷
│ Error: Failed to download module
│
│ Could not download module "default_tagging_policy_set" (infra/main.tf:90)
│ source code from
│ "git::ssh://git@github.com/myorg/terraform-azure-modules.git?ref=myref":
│ error downloading
│ 'ssh://git@github.com/myorg/terraform-azure-modules.git?ref=myref':
│ /usr/bin/git exited with 128: Cloning into
│ '.terraform/modules/infra.default_tagging_policy_set'...
│ Host key verification failed.
│ fatal: Could not read from remote repository.
│
│ Please make sure you have the correct access rights
│ and the repository exists.
│
╵
```

### TFLint Version

0.44.1

### Terraform Version

1.3.6

### Operating System

- [X] Linux
- [X] macOS
- [ ] Windows

**Comment by wata727** [comment]
TFLint does not download modules. I think you're probably misunderstanding. The errors are very similar to those reported by `terraform init`.

**Comment by arcjor** [comment]
So are you telling me that in a case like mine, where I have superlinter calling tflint, that it is also calling terraform init in these directories directly? And then subsequently calling tflint?

I'm fine with this not being accepted as a tflint bug, but I'd like to understand where to follow up.
